cmd/internal/find/primary/meta/timePredicate_test.go

```diff
@@ -0,0 +1,61 @@
+package meta
+
+import (
+	"testing"
+	"time"
+
+	"github.com/puppetlabs/wash/cmd/internal/find/params"
+	"github.com/puppetlabs/wash/cmd/internal/find/primary/numeric"
+	"github.com/stretchr/testify/suite"
+)
+
+type TimePredicateTestSuite struct {
+	ParserTestSuite
+}
+
+func (suite *TimePredicateTestSuite) SetupTest() {
+	params.StartTime = time.Now()
+}
+
+func (suite *TimePredicateTestSuite) TeardownTest() {
+	params.StartTime = time.Time{}
+}
+
+func (suite *TimePredicateTestSuite) TestErrors() {
+	suite.runTestCases(
+		nPETC("", `expected a \+, -, or a digit`, true),
+		nPETC("200", "expected a duration", true),
+		nPETC("+{", ".*closing.*}", false),
+	)
+}
+
+func (suite *TimePredicateTestSuite) TestValidInput() {
+	// Test the happy cases first
+	suite.runTestCases(
+		nPTC("+2h -size", "-size", addTST(-3*numeric.DurationOf('h'))),
```

Comment:
I think it'd be useful to have more negative tests where a valid value fails the predicate.

Reply:
Ok, can add those later this morning.